src/software/new_geom/vector_test.cpp

```diff
@@ -0,0 +1,228 @@
+#include "software/new_geom/vector.h"
+
+#include <gtest/gtest.h>
+
+#include "software/new_geom/point.h"
+
+TEST(CreateVectorTests, vector_default_constructor_test)
+{
+    Vector v = Vector();
+    EXPECT_EQ(0, v.x());
+    EXPECT_EQ(0, v.y());
+}
+
+TEST(CreateVectorTests, vector_from_angle)
+{
+    Vector v = Vector::createFromAngle(Angle::fromDegrees(30));
+    EXPECT_DOUBLE_EQ(sqrt(3) / 2, v.x());
+    EXPECT_DOUBLE_EQ(0.5, v.y());
+}
+
+TEST(CreateVectorTests, vector_specific_constructor_test)
+{
+    Vector v = Vector(-3, 4);
+    EXPECT_EQ(-3, v.x());
+    EXPECT_EQ(4, v.y());
+}
+
+TEST(CreateVectorTests, vector_copy_constructor_test)
+{
+    Vector v = Vector(5, -2);
+    Vector u = Vector(v);
+    EXPECT_EQ(5, u.x());
+    EXPECT_EQ(-2, u.y());
+}
+
+TEST(CreateVectorTests, vector_from_point_test)
+{
+    Point p  = Point(-3, 7);
+    Vector v = p.toVector();
+    EXPECT_EQ(-3, v.x());
+    EXPECT_EQ(7, v.y());
+}
+
+TEST(SetVectorTests, vector_set_x_and_y_test)
+{
+    Vector v = Vector();
+    v.setX(6);
+    v.setY(5);
+    EXPECT_EQ(6, v.x());
+    EXPECT_EQ(5, v.y());
+
+    v.set(2, 4);
+    EXPECT_EQ(2, v.x());
+    EXPECT_EQ(4, v.y());
+}
+
+TEST(VectorLogicTests, vector_len_and_lensq_test)
+{
+    Vector v = Vector(3, 4);
+    EXPECT_EQ(25, v.lensq());
+    EXPECT_EQ(5, v.len());
+}
+
+TEST(VectorLogicTests, norm_vector_from_vector_test)
+{
+    Vector u = Vector(3, 4);
+    Vector v = u.norm();
+    EXPECT_EQ(0.6, v.x());
+    EXPECT_EQ(0.8, v.y());
+}
+
+TEST(VectorLogicTests, norm_vector_from_vector_with_length_test)
+{
+    Vector u = Vector(3, 4);
+    Vector v = u.norm(2);
+    EXPECT_EQ(1.2, v.x());
+    EXPECT_EQ(1.6, v.y());
+}
+
```

Comment:
Could we add a test for taking the `norm()` of a vector that is just off of zero (but less then `EPSILON`)?

Reply:
updated